pkg/cloud/services/loadbalancer/loadbalancer.go

```diff
@@ -64,13 +63,18 @@ func (s *Service) ReconcileLoadBalancer(openStackCluster *infrav1.OpenStackClust
 		return err
 	}
 
-	// As mostly all LoadBalancer features are only supported on "amphora" we explicitly set the provider
-	// in the LoadBalancer create call to make sure to get the desired features - even if multiple providers exist.
-	var lbProvider string
-	for _, v := range providers {
-		if v.Name == defaultLoadBalancerProvider {
-			lbProvider = v.Name
-			break
+	// Choose the selected provider if it is set in cluster spec, if not, omit the field and Octavia will use the default provider.
+	lbProvider := ""
+	if openStackCluster.Spec.APIServerLoadBalancer.Provider != "" {
+		for _, v := range providers {
+			if v.Name == openStackCluster.Spec.APIServerLoadBalancer.Provider {
+				lbProvider = v.Name
+				break
+			}
+		}
+		if lbProvider == "" {
+			record.Warnf(openStackCluster, "OctaviaProviderNotFound", "Provider specified for Octavia not found.")
+			record.Eventf(openStackCluster, "OctaviaProviderNotFound", "Provider specified for Octaiva not found, using the default provider.")
```

Comment:
There's an "Octaiva" typo.

Reply:
Also now looking at this closer, I think I'd include the information about which provider is missing:

`"Provider %s specified for API LB not found, using the default Octavia provider.", openStackCluster.Spec.APIServerLoadBalancer.Provider`